src/cljs_http/core.cljs

```diff
@@ -29,15 +29,28 @@
           h-val (vals headers)]
     (.set (.-headers xhr) h-name h-val)))
 
+(defn apply-response-type!
+  "Takes an XhrIo object and sets response-type if not nil."
+  [xhr response-type]
+  (.setResponseType xhr
+   (case response-type
+     :arraybuffer XhrIo.ResponseType.ARRAY_BUFFER
+     :blob        XhrIo.ResponseType.BLOB
+     :document    XhrIo.ResponseType.DOCUMENT
+     :text        XhrIo.ResponseType.TEXT
+     :default     XhrIo.ResponseType.DEFAULT
+     nil          XhrIo.ResponseType.DEFAULT)))
+
```

Comment:
I chose this explicitly such that when users make a typo or ask for non-supported response type, they get an error from `case`. Do you think it's good idea to swallow it with the default `case` handler?